src/platform/packages/shared/response-ops/rule_form/src/rule_form_screen_context/rule_form_screen_context.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { createContext, useState } from 'react';
+
+/*
+ * A generic wrapper for keeping track of which screens to show on top of the Rule Form
+ * This provides logic that works on both the Rule Page, which displays these screens in a modal,
+ * and the Rule Flyout, which displays these screens by replacing the entire content of the flyout.
+ */
+const initialRuleFormScreenContextState = {
```

Comment:
I'm unsure what "screen" is referring to. I would think "element" might be fitting too, but not sure. Could you please elaborate a little bit on the reasoning behind "screen"?

Reply:
Idk naming things is hard. I don't think it's an "element" because it could either be a modal or a full replacement of the content of the flyout, and there isn't a specific DOM element that corresponds to in both cases. "View" maybe? Remember when it was 2015 and we had model-view-controllers?